<launch>

    <!-- generate disparity map and confidence map from disparity_raw  -->
    <include file="$(find ti_viz_nodes)/launch/viz_disparity.launch"/>

    <!-- color conversion for input image_raw for visualization -->
    <node pkg="ti_viz_nodes" type="viz_color_conv_yuv2rgb" name="viz_color_conv_yuv2rgb_node_rect" output="screen" args="" required="true">
        <param name="width"             value="1280"/>
        <param name="height"            value="720"/>
        <param name="input_yuv_topic"   value="camera/right/image_rect_nv12"/>
        <param name="output_rgb_topic"  value="camera/right/image_rect_color"/>
        <param name="yuv_format"        value="YUV420"/>
        <param name="yuv420_luma_only"  value="false"/>
    </node>

    <!-- viz_objdet_range  -->
    <node pkg="ti_viz_nodes" type="viz_objdet_range" name="viz_objdet_range" output="screen" args="" required="true">
        <param name="fused_obj_range_topic"    value="camera/fused_objdet_range"/>
        <param name="rectified_image_topic"    value="camera/right/image_rect_color"/>
        <param name="output_image_topic"       value="vision_cnn/out_image"/>
        <rosparam>
            box_color_rgb:     [102, 255, 102]
            patch_color_rgb:   [255, 128, 0]
            text_color_rgb:    [244,  35, 232]
            text_bg_color_rgb: [120, 120, 120]
        </rosparam>
    </node>

    <!-- rviz -->
    <!-- <node pkg="rviz" type="rviz" name="rviz_sde" args="-d $(find ti_viz_nodes)/rviz/objdet_range.rviz" /> -->
    <node pkg="rviz" type="rviz" name="rviz_sde" args="-d $(find ti_viz_nodes)/rviz/objdet_range_disp_pcl.rviz" />
    <!-- <node pkg="rviz" type="rviz" name="rviz_sde" args="-d $(find ti_viz_nodes)/rviz/objdet_range_disp_pcl_scan.rviz" /> -->

</launch>
